code.js

```diff
@@ -15,3 +15,9 @@ function fib(s) {
 }
 console.log(fib(10));
 // get help from TA
+
+//Invariant
+/**
+ * What I think is always true is the new array element is the sum of the previous two numbers,
```

Comment:
And how does this help you to prove that the function works as intended?

Reply:
When s is 0, 1, or 2, we set the result array as [0], [0,1], or [0,1,1] respectively. We then recursively call the fib(s-1) function to calculate the sum of the previous two items and store it as a new result.

Reply:
And how is that related to the Fibonacci numbers?

Reply:
Continue to add every two previous counting results and then push into the array to get the Fibonacci numbers series.